pallets/compliance-manager/src/lib.rs

```diff
@@ -735,9 +843,22 @@ impl<T: Trait> Module<T> {
                 asset_compliance_with_results.result = true;
             }
         }
+        // Aggregating the result with the implicit requirement result.
+        asset_compliance_with_results.result =
+            Self::compute_final_result(&asset_compliance_with_results);
         asset_compliance_with_results
     }
 
+    fn compute_final_result(asset_compliance: &AssetComplianceResult) -> bool {
+        let compliance_result = asset_compliance.result;
+        asset_compliance
+            .implicit_requirements
+            .as_ref()
+            .map_or(compliance_result, |imp_result| {
+                imp_result.result && compliance_result
```

Comment:
If `compliance_result == false` then the whole computation using `asset_compliance` can be skipped because the result of `&&` is always going to be `false`.

Reply:
Yes, but we want explicitly store the result of the condition within it so UI guys know where it fails exactly.